File: 000/FridgyKeyApp/FridgyKeyApp.Models/Models/Product.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FridgyKeyApp.DAL
{
    [Table("Product")]
    public partial class Product
    { 
        public Product()
        {
            FridgeProducts = new HashSet<FridgeProduct>();
        }

        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int Id { get; set; }
         

        [StringLength(50)]
        public string Name { get; set; }

        [StringLength(255)]
        public string Description { get; set; }

        [StringLength(255)]
        public string ImgURL { get; set; }


        public virtual ICollection<FridgeProduct> FridgeProducts { get; set; }

        public virtual ApplicationUser User { get; set; }
    }
}

File: aspnetcore-angular/000my/000/UI/src/app/models/Product.ts
import { UserDetail } from './UserDetail';
import { ProductInfo } from './ProductInfo';

export class Product {
    Id: number;
    Name: string;
    Description: string;
    ImgURL: string;
    ProdInfo: ProductInfo;
    User: UserDetail;
}

/*
 public int Id { get; set; }

        [StringLength(450)]
        public virtual string UserId { get; set; }

        [StringLength(450)]
        public string Name { get; set; }

        [StringLength(450)]
        public string Description { get; set; }

        [StringLength(255)]
        public string ImgURL { get; set; }


        public virtual ICollection<FridgeProduct> FridgeProducts { get; set; }

        //public virtual ICollection<Ingredient> Ingredients { get; set; }

        public virtual ProductInfo ProdInfo { get; set; }

        public virtual ApplicationUser User { get; set; }
        */